### PR TITLE
chore: capturing user_transform_test_errors stat

### PR DESCRIPTION
**Pull request by kanishkkatara**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.
https://linear.app/rudderstack/issue/DAT-1394/capture-alerts-for-rate-limit-exceed

## What is the related Linear task?

Resolves INT-XXX

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3543/test-report.html

File: src/services/userTransform.ts [hunk 1]
```diff
@@ -231,6 +231,12 @@ export class UserTransformService {
       response.body = {
         error: extractStackTraceUptoLastSubstringMatch(error.stack, SUPPORTED_FUNC_NAMES),
       };
+      const metaTags = getTransformationMetadata(events[0]?.metadata);
```
Comment (Jayachand) [inline src/services/userTransform.ts:234]: modify config-backend caller to reflect these tags.
/test call events metadata is constant and they do not have transformation or workspace details

Comment (Jayachand) [reply src/services/userTransform.ts:234]: Or extract workspaceId from trRevcode which will have better hit rate

File: src/services/userTransform.ts [hunk 2]
```diff
@@ -231,6 +231,12 @@ export class UserTransformService {
...
+
+      stats.counter('user_transform_test_errors', events.length, {
+        status: error.statusCode,
+        ...metaTags,
+      });
```
Comment (abhimanyubabbar) [inline src/services/userTransform.ts:239]: Why not write this in finally block and capture every status code, we can always filter on statuscode != 200 but atleast we will know how many calls to test transform by what workspaceId are being sent. Wdyt ?!

Comment (abhimanyubabbar) [reply src/services/userTransform.ts:239]: we can name the stat `user_transform_test_count_total`

**Review by abhimanyubabbar — CHANGES_REQUESTED** [review]

File: src/services/userTransform.ts (hunk 2 above)
Comment (Jayachand) [reply src/services/userTransform.ts:239]: Main concern right now is errors. Having total count also makes sense, it might be useful in the future

Comment (kanishkkatara) [reply src/services/userTransform.ts:239]: Sounds good, making that change

**Review by abhimanyubabbar — APPROVED** [review]

**Pull request merged**